rpc/src/v1/types/sync.rs

```diff
@@ -32,14 +33,60 @@ pub struct SyncInfo {
 }
 
 /// Peers info
-#[derive(Default, Debug, Serialize, PartialEq)]
+#[derive(Default, Debug, Serialize)]
 pub struct Peers {
 	/// Number of active peers
 	pub active: usize,
 	/// Number of connected peers
 	pub connected: usize,
 	/// Max number of peers
 	pub max: u32,
+	/// Detailed information on peers
+	pub peers: Vec<PeerInfo>
+}
+
+/// Peer connection information
+#[derive(Default, Debug, Serialize)]
+pub struct PeerInfo {
+	/// Public node id
+	pub id: Option<String>,
+	/// Node client ID
+    pub name: String,
+	/// Capabilities
+	pub caps: Vec<String>, 
+	/// Network information
+	pub network: PeerNetworkInfo,
+	/// Protocols information
+	pub protocols: PeerProtocolsInfo
+}
+
+/// Peer network information
+#[derive(Default, Debug, Serialize)]
+pub struct PeerNetworkInfo {
+	/// Remote endpoint address
+	#[serde(rename="remoteAddress")]
+    pub remote_address: String,
+	/// Local endpoint address
+	#[serde(rename="localAddress")]
+    pub local_address: String
+}
+
+/// Peer protocols information
+#[derive(Default, Debug, Serialize)]
+pub struct PeerProtocolsInfo {
+	/// Ethereum protocol information
+	pub eth: PeerEthereumProtocolInfo
+}
```

Comment:
@svyatonik I agree that strictly typed is better. For now, I think it can be left as is, since the only peers we'll find will be connected to `eth`, but it should be documented as a nullable/optional field in the JSON structure so we can accommodate future subprotocols without breaking backwards compatibility.